spacy/cli/package.py

```diff
@@ -109,6 +110,24 @@ def package(
             ", ".join(meta["requirements"]),
         )
     if name is not None:
+        if not name.isidentifier():
+            msg.fail(
+                f"Model name ('{name}') is not a valid python identifier. "
+                "This is required to correctly package the pipeline so it can be imported as a module.",
+                "We recommend names that use ASCII A-Z, a-z, _ (underscore), "
+                "and 0-9 (except as the first character). "
+                "For specific details see: https://docs.python.org/3/reference/lexical_analysis.html#identifiers",
+                exits=1,
+            )
+        if not _is_permitted_package_name(name):
+            msg.fail(
+                f"Model name ('{name}') is not a permitted package name. "
+                "This is required to correctly install the pipeline after packaging.",
```

Comment:
@adrianeboyd Is this the right specific reason for this error? I could use your help on the wording here. My understanding is that the package should still be able to be created, but it wouldn't be `pip` install-able since it's not a valid package name and it would be normalized if they uploaded it to PyPI?

Reply:
Well, we wouldn't usually upload model packages to pypi, so that's not the direct concern. It's a concern that you can package and install it, but then you can't load it with `spacy.load`.

Reply:
I'm not sure whether a least a few of the edge cases with `spacy.load` are our fault and not some packaging internals, but the overall goal is that you want to install something you can load normally.